quic/s2n-quic-transport/src/path/manager.rs

```diff
@@ -249,44 +249,25 @@ impl<Config: endpoint::Config> Manager<Config> {
     ) -> Result<(Id, AmplificationOutcome), DatagramDropReason> {
         let valid_initial_received = self.valid_initial_received();
 
-        if let Some((id, path)) = self.path_mut(path_handle) {
-            let source_cid_changed = datagram
-                .source_connection_id
-                .is_some_and(|scid| scid != path.peer_connection_id && valid_initial_received);
-
-            if source_cid_changed {
-                //= https://www.rfc-editor.org/rfc/rfc9000#section-7.2
-                //# Once a client has received a valid Initial packet from the server, it MUST
-                //# discard any subsequent packet it receives on that connection with a
-                //# different Source Connection ID.
-
-                //= https://www.rfc-editor.org/rfc/rfc9000#section-7.2
-                //# Any further changes to the Destination Connection ID are only
-                //# permitted if the values are taken from NEW_CONNECTION_ID frames; if
-                //# subsequent Initial packets include a different Source Connection ID,
-                //# they MUST be discarded.
-
-                return Err(DatagramDropReason::InvalidSourceConnectionId);
-            }
-
-            // Update the address if it was resolved
-            //
-            // NOTE: We don't update the server address since this would cause the client to drop
-            // packets from the server.
-
-            //= https://www.rfc-editor.org/rfc/rfc9000#section-9
-            //# If a client receives packets from an unknown server address, the client MUST discard these packets.
-
+        let matched_path = if handshake_confirmed {
+            self.path_mut(path_handle)
+        } else {
             //= https://www.rfc-editor.org/rfc/rfc9000#section-9
-            //# If the peer sent the disable_active_migration transport parameter, an endpoint also MUST NOT send
-            //# packets (including probing packets; see Section 9.1) from a different local address to the address
-            //# the peer used during the handshake, unless the endpoint has acted on a preferred_address transport
-            //# parameter from the peer.
-            if Config::ENDPOINT_TYPE.is_client() {
-                path.handle.maybe_update(path_handle);
-            }
+            //# The design of QUIC relies on endpoints retaining a stable address
+            //# for the duration of the handshake.  An endpoint MUST NOT initiate
+            //# connection migration before the handshake is confirmed, as defined
+            //# in section 4.1.2 of [QUIC-TLS].
+
+            // NOTE: while we must not _initiate_ a migration before the handshake is done,
```

Comment:
Do we want some kind of event that we received a packet from a weird source address but we're going to process it anyways?

Reply:
Good question... I think it could be useful for overall insights into what we're doing. But in terms of metrics, there really isn't anything actionable you can do since it's not directly controllable when addresses change.

But since it's easy to add events I can add one.

Reply:
added in https://github.com/aws/s2n-quic/pull/2463/commits/63b82c8efc1c1385c44e1f9fbb531b4a025de815